lib/enkf/value_export.cpp

```diff
@@ -46,86 +48,147 @@ static void backup_if_existing(const char * filename) {
+
+
+            if (std::next(iterValues) != subMap.end())
+                fprintf(stream, ",");
+            fprintf(stream, "\n");
+        }
+
+        fprintf(stream, "},\n");
+
+    }
+
+}
+
+void generate_comosite_keys(const value_export_type * value, FILE * stream)
+{
+    for (auto iterMaps = value->values.begin(); iterMaps!= value->values.end(); ++iterMaps   ) {
+        std::string key = (*iterMaps).first;
+        std::map<std::string,double> subMap = (*iterMaps).second;
+
+        for (auto iterValues = subMap.begin(); iterValues != subMap.end(); ++iterValues) {
+
+            std::string subkey = (*iterValues).first;
+
+            double double_value = (*iterValues).second;
+            if (isnan(double_value))
+                fprintf(stream, "\"%s\" : NaN", key.c_str());
+            else
+                fprintf(stream, "\"%s:%s\" : %g", key.c_str(), subkey.c_str(), double_value);
+
+            if (std::next(iterValues) != subMap.end()) {
+                fprintf(stream, ",");
+                fprintf(stream, "\n");
+            }
+        }
+
+
+        if (std::next(iterMaps) != value->values.end())
+            fprintf(stream, ",");
+
+        fprintf(stream, "\n");
+
+    }
+
 }
 
 void value_export_json(const value_export_type * value) {
-  char * filename = util_alloc_filename( value->directory , value->base_name, "json");
-  backup_if_existing(filename);
-  const int size = double_vector_size( value->values );
-  if (size > 0) {
-    FILE * stream = util_fopen( filename , "w");
+  std::string filename = value->directory +"/" + value->base_name + ".json";
+  backup_if_existing(filename.c_str());
+
+  if (!value->values.empty()) {
+    FILE * stream = util_fopen( filename.c_str() , "w");
     fprintf(stream, "{\n");
-    for (int i=0; i < size; i++) {
-      const char * key          = stringlist_iget( value->keys, i );
-      double double_value       = double_vector_iget( value->values, i );
-      if (isnan(double_value))
-        fprintf(stream,"\"%s\" : NaN", key);
-      else
-        fprintf(stream, "\"%s\" : %g", key, double_value);
-
-      if (i < (size - 1))
-        fprintf(stream, ",");
-      fprintf(stream,"\n");
-    }
+    generate_hirarchical_keys(value, stream);
+    generate_comosite_keys(value, stream);
     fprintf(stream, "}\n");
     fclose( stream );
   }
-  free( filename );
+
 }
 
 void value_export(const value_export_type * value) {
   value_export_txt( value );
   value_export_json( value );
 }
 
-void value_export_append( value_export_type * value, const char * key , double double_value) {
-  stringlist_append_copy( value->keys, key );
-  double_vector_append( value->values, double_value );
-}
 
+void value_export_append(value_export_type * value, const std::string key, const std::string subkey, double double_value){
+
+  if(value->values.find(key) == value->values.end()){
+    value->values.insert(std::make_pair(key, std::map<std::string, double>()));
```

Comment:
You can just do: `value->values[key][subkey] = value`. The `[]` operator on a map will auto-create an instance and return a reference to it.